File: d66-social/src/main/java/com/eroldmr/d66/subreddit/comment/Comment.java
package com.eroldmr.d66.subreddit.comment;

import com.eroldmr.d66.appuser.AppUser;
import com.eroldmr.d66.subreddit.post.Post;
import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;

import static javax.persistence.FetchType.LAZY;

/**
 * @author Mompati 'Patco' Keetile
 * @created 19-12-2022 @ 11:52
 */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder(builderMethodName = "NewComment")
@Entity
public class Comment {
  @Id
  @SequenceGenerator(name = "comment_id_seq", sequenceName = "comment_id_seq", allocationSize = 1)
  @GeneratedValue(generator = "comment_id_seq", strategy = GenerationType.IDENTITY)
  private Long id;
  @NotEmpty
  private String text;
  @ManyToOne(fetch = LAZY)
  @JoinColumn(name = "postId", referencedColumnName = "postId")
  private Post post;
  @ManyToOne(fetch = LAZY)
  @JoinColumn(name = "userId", referencedColumnName = "userId")
  private AppUser appUser;
}

File: d66-social/src/main/kotlin/com/eroldme/d66/subredit/comment/Comment.kt
package com.eroldme.d66.subredit.comment

import com.eroldme.d66.subredit.post.Post
import com.eroldme.d66.user.ApplicationUser
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor
import java.time.Instant
import javax.persistence.*
import javax.validation.constraints.NotEmpty

/**
 * @author Mompati 'Patco' Keetile
 * @created 06-12-2022 @ 13:57
 *
 */
@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
class Comment(
    @Id
    @SequenceGenerator(name = "comment_id_seq", sequenceName = "comment_id_seq", allocationSize = 1)
    @GeneratedValue(generator = "comment_id_seq", strategy = GenerationType.IDENTITY)
    private val id: Long,
    @NotEmpty
    private val text: String,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "postId", referencedColumnName = "postId")
    private val post: Post,
    private val createdDate: Instant,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userId", referencedColumnName = "userId")
    private val applicationUser: ApplicationUser
)